packages/cozy-harvest-lib/src/services/softDeleteOrRestoreAccounts.js

```diff
@@ -0,0 +1,85 @@
+import logger from './logger'
+import {
+  decryptString,
+  getOrganizationKey,
+  fetchAccountsForCipherId,
+  fetchKonnectorFromAccount,
+  fetchTriggersFromAccount,
+  updateAccountsAuth
+} from './utils'
+import get from 'lodash/get'
+import { ensureTrigger } from '../connections/triggers'
+import { translate as t } from 'cozy-ui/transpiled/react/I18n'
+
+const makeDecrypt = (vaultClient, orgKey) => encryptedVal =>
+  decryptString(encryptedVal, vaultClient, orgKey)
+
+const softDeleteOrRestoreAccounts = async (
+  cozyClient,
+  vaultClient,
+  cipherDoc
+) => {
+  const cipherId = get(cipherDoc, '_id')
+
+  logger.debug(`Fetching accounts for cipher ${cipherId}...`)
+  const accounts = await fetchAccountsForCipherId(cozyClient, cipherId)
+  if (accounts.data.length < 1) {
+    logger.debug('No account found for this cipher.')
+    return
+  }
+  const orgKey = await getOrganizationKey(cozyClient, vaultClient)
+  const decrypt = makeDecrypt(vaultClient, orgKey)
+
+  const encryptedPassword = get(cipherDoc, 'login.password')
+  const encryptedUsername = get(cipherDoc, 'login.username')
+
+  const encryptedFields = get(cipherDoc, 'fields') || []
+  const decryptedFields = {}
+  for (const encryptedField of encryptedFields) {
+    const fieldName = await decrypt(encryptedField.name)
+    const fieldValue = await decrypt(encryptedField.value)
+    decryptedFields[fieldName] = fieldValue
+  }
+
+  const decryptedPassword = await decrypt(encryptedPassword)
+  const decryptedUsername = await decrypt(encryptedUsername)
```

Comment:
Nit: All this decrypting logic could be extracted as a function returning a decrypted object.

Reply:
Right, but I prefer not moving it, because it will need some refactoring (decrypt is only available here) and adapt the tests

Reply:
I do not see why you need to adapt the tests 🤔  This function takes 70 lines, it starts to be big. Moving the creation of the decrypt function outside function would make it easier to see the bigger picture of this function.